lighthouse-core/gather/gather-runner.js

```diff
@@ -215,6 +215,31 @@ class GatherRunner {
     return new LHError(LHError.errors.CHROME_INTERSTITIAL_ERROR);
   }
 
+  /**
+   * Returns an error if we try to load a non-HTML page.
+   * @param {LH.Artifacts.NetworkRequest|undefined} mainRecord
+   * @return {LH.LighthouseError|undefined}
+   */
+  static getDocTypeError(mainRecord) {
+    // MIME types are case-insenstive
+    const HTML_MIME_REGEX = /^text\/html$/i;
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types, "MIME types are case-insensitive but are traditionally written in lowercase". I am unsure if the check needs to be this rigorous/if a constant 'text/html' is good enough.

Reply:
would `mainRecord.resourceType === NetworkRequest.RESOURCE_TYPES.Document` work too? perhaps not for xml docs...

Reply:
just double checked and xml docs would indeed pass (no error) if we used that condition

Reply:
does the mime type in the network records return Chrome's inferred mime type or just what was set on the header?

we should test it, but since the failure mode is so drastic, I would prefer to keep the no mimeType case passing (i.e. only reject documents that have an explicit non-HTML mimeType set)